src/charm.py

```diff
@@ -141,22 +143,46 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
         event.set_results(results)  # type: ignore[arg-type]
         self.model.unit.status = ops.ActiveStatus()
 
+    def _get_random_password(self) -> str:
```

Comment:
What is the reason for putting this on the charm class please? Perhaps a separate module?

Reply:
Moved to User class.